### ISSUE TITLE
Inspection not understand correctly checking value for null.

### ISSUE THREAD

**Issue by haegemon** [kalessil/phpinspectionsea#851]
Inspection not understand correctly checking value for null.

Let's see 3 examples:
```
<?php 
/** @var object|null $a */
$a = $someService->find(2);
if (\is_null($a)) {
    //do some
}
$a->getId() // inspection say that $a might be null, but this is wrong
```
If we say directly types inspection ignore that. But phpstorm hinting understand
```
<?php 
/** @var object|null $a */
$a = $someService->find(2);
if (\is_null($a)) {
    //do some
}
/** @var object $a */
$a->getId() // inspection say that $a might be null, but this is wrong. Phpstorm hint write that $a is only object
```
Example with if:
```
<?php 
/** @var object|null $a */
$a = $someService->find(2);
if ($a) {
    $a->getId() // inspection and Phpstorm say that $a is object oly
}
```
As I understand inspection not prepare for changing type of variable without assignment

**Comment by kalessil** [comment]
That's limitations of the inspection: https://github.com/kalessil/phpinspectionsea/blob/master/docs/probable-bugs.md#null-pointer-exceptions-prevention

If you try using a direct null comparison, the warning should disappear. And considering type-respecification: I'm avoiding PhpDoc in the analysis as much as possible and rely more on the workflow (it's pretty easy to fool IDE, but we are catching such cases =)).

**Comment by kalessil** [comment]
UPD: type change is considered, but I don't remember if it's in the Extended or Ultimate plugin.
